textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/TrigramBooleanQuery.java

```diff
@@ -0,0 +1,172 @@
+package edu.uci.ics.textdb.dataflow.regexmatch;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.StringJoiner;
+
+import edu.uci.ics.textdb.common.constants.DataConstants;
+
+/**
+ * 
+ * Trigram Query of OR and AND
```

Comment:
We want the "3" in "trigram" to be a customizable parameter, it will be better to call it "Gram" instead of "Trigram".